developer-guide.md

```diff
@@ -0,0 +1,353 @@
+
+```css
+.site-header {
+  page-transition-tag: side-header;
+  /* Paint containment is required */
+  contain: paint;
+}
+```
+
+This changes how the page is captured. When `transition.start(callback)` is called:
+
+1. For every rendered element that has a `page-transition-tag`:
+   1. Extract it from the page, and create a `::page-transition-container` for it, adding in a `::page-transition-outgoing-image`.
+1. Do the same for the remainder of the page.
+1. Call `callback`, and wait for its returned promise to resolve.
+1. For every rendered element that has a `page-transition-tag`:
+   1. Extract it from the page.
+   1. If there's already a `::page-transition-container` for a `page-transition-tag` of this name, use it, otherwise create a new one.
+   1. Adding in a `::page-transition-incoming-image`.
+
+If our header exists on both sides of the DOM change, we'll now have a structure like this:
+
+```html
+<special-top-layer>
+  <container(root)>
+    <image-wrapper(root)>
+      <outgoing-image(root) />
+      <incoming-image(root) />
+    </image-wrapper(root)>
+  </container(root)>
+
+  <container(site-header)>
+    <image-wrapper(site-header)>
+      <outgoing-image(site-header) />
+      <incoming-image(site-header) />
+    </image-wrapper(site-header)>
+  </container(site-header)>
+</special-top-layer>
+```
+
+Although, it's possible to have one of these structures that doesn't have an incoming/outgoing image. For example, if the element was only there before the DOM change, or only there after the DOM change.
+
+In a similar way that we were using `::page-transition-outgoing-image(root)` to target the outgoing image of the root, we can use `::page-transition-outgoing-image(site-header)` to target the outgoing image of the site-header, and animate it separately.
+
+The default animation animates the `width` and `height` of the `::page-transition-container` from its before size to its after size, while animating a `transform` to move it from its before position to its after position. If the header is unchanged, it'll appear fixed in place while the rest of the content (the root) slides as we specified earlier.
+
+### Flat transition structures
+
+Each captured element, along with the root, will be a direct child of the special top layer. This means you can move elements between containers, even if they have `overflow: hidden` or some other form of clipping in the real DOM.
+
+We plan to add a feature to allow one transition container to be nested within another, but that isn't currently implemented.
+
+## Animation synchronization gotcha
+
+Right now, there's a delay between Chrome creating the outgoing image element, and the incoming image element. That means the styles for the outgoing image element will apply sooner, which in turn means its animation will start sooner, and apply out of sync with the incoming image element.
+
+This is considered a bug (well, a design error), but you can work around it:
```

Comment:
Sounds like we should make crbug.com/1310798 OT blocking? :)

Reply:
It's workaroundble at least, but yeah it'd be good to have it fixed.